src/hotspot/share/gc/shared/parallelCleaning.cpp

```diff
@@ -104,7 +104,7 @@ bool KlassCleaningTask::claim_clean_klass_tree_task() {
     return false;
   }
 
-  return Atomic::cmpxchg(&_clean_klass_tree_claimed, 0, 1) == 0;
+  return !Atomic::cmpxchg(&_clean_klass_tree_claimed, false, true);
```

Comment:
Generally, I prefer to see `Atomic::cmpxchg(&_clean_klass_tree_claimed, false, true) == false`, because `false` here is not about the logical boolean expression, but about the "old" value`false`, which just happens to be a boolean.

Reply:
I'll ask internally first too, and then fix all of these occurrences at once if needed.

That particular occurrence may go away soon too.